devmapper: Make sure device is a thin pool device

rhvgoyal opened the pull request:
Fixes #16092 

Right now we check for the existence of device but don't make sure it is
a thin pool device. We assume it is a thin pool device and call poolStatus()
on the device which returns an error EOF. And that error does not tell
anything.

So before we reach the stage of calling poolStatus() make sure we are working
with a thin pool device otherwise error out.

Signed-off-by: Vivek Goyal vgoyal@redhat.com

LK4D4 commented on daemon/graphdriver/devmapper/deviceset.go:
```diff
@@ -897,6 +897,33 @@ func (devices *DeviceSet) createBaseImage() error {
+
+	_, _, deviceType, _, err := devicemapper.GetStatus(thinPoolDevice)
+	if err != nil {
+		return true, fmt.Errorf("devmapper: GetStatus() on device %s failed: %v", thinPoolDevice, err)
```
this signature slightly bother me, let's return false(default bool value) on errors, that value shouldn't be used in case of error anyway.

rhvgoyal replied:
Ok, will change it.

rhvgoyal committed:
devmapper: Make sure device is a thin pool device

Right now we check for the existence of device but don't make sure it is
a thin pool device. We assume it is a thin pool device and call poolStatus()
on the device which returns an error EOF. And that error does not tell
anything.

So before we reach the stage of calling poolStatus() make sure we are working
with a thin pool device otherwise error out.

Signed-off-by: Vivek Goyal <vgoyal@redhat.com>

rhvgoyal commented:
@LK4D4 Did the change. PTAL.

LK4D4 commented:
LGTM

rhvgoyal commented:
@vbatts Please have a look.

vbatts commented:
LGTM
